packages/app/src/templates/services/template.service.ts

```diff
@@ -42,4 +49,48 @@ export class TemplateService {
       }
     }
   }
+
+  public async deleteTemplates(
+    templateIds: string[]
+  ): Promise<RequestResponse<string[]>> {
+    const results = templateIds.map(async (id) => {
+      const { status } = await this.deviceService.request({
+        endpoint: Endpoint.Messages,
+        method: Method.Delete,
+        body: {
+          category: PureMessagesCategory.template,
+          templateID: Number(id),
+        },
+      })
+      return {
+        status,
+        id,
+      }
+    })
+
+    const errorIds = (await Promise.all(results))
+      .filter(({ status }) => status === RequestResponseStatus.Error)
+      .map(({ id }) => id)
+    const successIds = (await Promise.all(results))
+      .filter(({ status }) => status === RequestResponseStatus.Ok)
+      .map(({ id }) => id)
+
+    if (errorIds.length > 0) {
+      successIds.forEach((id) => this.templateRepository.delete(id))
+
+      return {
+        status: RequestResponseStatus.Error,
+        error: {
+          message: "Delete template: Something went wrong",
+          data: errorIds,
```

Comment:
It might conflict a little bit about with my above comment (the one with `DeleteTemplateRequestResponse`) but what do you think about returning something like: 
```
      return {
        status: RequestResponseStatus.Error,
        error: {
          message: "Delete template: Something went wrong",
          data: { errorIds, successIds },
        },
      }
```

It would make your life easier later (in `delete-templates.action.ts` you would not need to filter those values). What do you think about it?

Reply:
I included this in the changes from your previous suggestion in [5ca7841](https://github.com/mudita/mudita-center/pull/752/commits/5ca7841b5759555b1bd575843e9a4c7a8ea57487)